src/app/components/room/room.component.ts

```diff
@@ -1,15 +1,99 @@
 import { Component, OnInit } from '@angular/core';
+import {readPost, User, writePost } from 'src/app/shared-module/models/user';
+import { LoginService } from 'src/app/services';
+import { Constants } from 'src/app/shared-module/constants/constants';
+import { Router } from '@angular/router';
+import { ToasterService } from 'src/app/core-module/services';
+import { isJSDocThisTag } from 'typescript';
+
 
 @Component({
   selector: 'app-room',
   templateUrl: './room.component.html',
   styleUrls: ['./room.component.scss']
 })
 export class RoomComponent implements OnInit {
+  messageInput: string = "";
+  posts: readPost[];
 
-  constructor() { }
+  constructor(private loginService: LoginService, private toaster: ToasterService, private router: Router) {
+    this.getPosts();
+   }
 
   ngOnInit(): void {
   }
 
+  getPosts(): void{
+    this.loginService.testGetPosts().subscribe(
+      (post:readPost[])=>this.posts = post,
+      // (post: Post)=>{console.log("I GOT THIS POST", post),
+      (err) => this.handleError(err.error.Error[0]),
+      ()=>{console.log(this.posts)}
+    );
+    // console.log(this.posts);
+  }
+
+  getUser(userID:number): void{
+    this.loginService.getUserForPost(userID).subscribe(
+      (user:User)=>console.log(user),
+      (err) => this.handleError(err.error.Error[0])
+    );
+  }
+
+  get displayUser():String {
+    return String(this.getUser(this.UserFromSessionStorage));
+  }
+
+  postPost(): void{
+    if(this.canPost){
+
+      const post: writePost = {
+      message: this.messageInput,
+      rating: 0,
+      createdBy: this.UserFromSessionStorage,
+      createdDate: new Date()
+      }
+      this.loginService.addPost(post).subscribe(
+        (post:readPost)=>this.posts.push(post),
+        (err) => this.handleError(err.error.Error[0])
+      );
+
+      this.messageInput = "";
+    }
+  }
+
+  get UserFromSessionStorage(): number{
+    return JSON.parse(sessionStorage.getItem("user")).userID;
```

Comment:
Could you change this to use the "user" constant from our Constants class (like stated above). 

`Constants.sessionKeys.user`